/*
Copyright 2017 InitialDLab

Permission is hereby granted, free of charge, to any person obtaining a copy of
this software and associated documentation files (the "Software"), to deal in
the Software without restriction, including without limitation the rights to
use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
of the Software, and to permit persons to whom the Software is furnished to do
so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/
// Protocal buffers which are used to store the state of the server
// so if the server crashes it has a chance to recover without the user
// having to issue to build command for every structure the database
// was tracking.
//
// Robert Christensen (robertc@cs.utah.edu)

syntax = "proto3";

import "sampling_api.proto";

package ServerState;

message SamplingStructureInformation
{
	/* the name of the index */
	string name = 1;

	/* the file (and path) backing the data structure */
	string filename = 2;

	/* the type of payload this data structure was built to support */
	serverProto.SampleStructureType type = 3;

	/* when this data structure was built */
	int64 time_built = 4;

	/* the time the samples for the structure were built */
	int64 time_samples_built = 5;
}

message SamplingStructureRepoDump
{
	/* The dump which contains the data for the dump */
	repeated SamplingStructureInformation data = 1;

	/* when this dump was created (so if we have more than one we know which is more recent) */
	int64 time_saved = 2;
}
